crawl-ref/source/spl-book.cc

```diff
@@ -874,6 +870,14 @@ static bool _learn_spell_checks(spell_type specspell, bool wizard = false)
 */
 bool learn_spell(spell_type specspell, bool wizard)
 {
+    if (god_loathes_spell(specspell, you.religion))
```

Comment:
Consider extracting these six lines into their own function that returns a string. Then when you call it here you won't have to duplicate the `mprf(MSGCH_WARN, ...)` prelude, and you can avoid interleaving side effects with computation.